fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneScanAllEntriesTest.java

```diff
@@ -0,0 +1,220 @@
+                Arguments.of(false, true, true, false, Set.of(keyFor(2, 1001L))), // non-synthetic, ALL_ENTRIES, grouped, no EmptyDoc
+                Arguments.of(false, true, false, true, Set.of(keyFor(1, 1001L), keyFor(2, 1002L), keyFor(2, 1003L))), // non-synthetic, ALL_ENTRIES, non-grouped, with EmptyDoc
+                Arguments.of(false, true, false, false, Set.of(keyFor(1, 1001L), keyFor(2, 1002L))), // non-synthetic, ALL_ENTRIES, non-grouped, no EmptyDoc
+
+                Arguments.of(true, false, true, true, Set.of(syntheticKeyFor(2, 1001L, 1000L))), // synthetic, ONE_TERM, grouped, with EmptyDoc
+                Arguments.of(true, false, true, false, Set.of(syntheticKeyFor(2, 1001L, 1000L))), // synthetic, ONE_TERM, grouped, no EmptyDoc
+                Arguments.of(true, false, false, true, Set.of(syntheticKeyFor(1, 1001L, 1000L), syntheticKeyFor(2, 1002L, 999L))), // synthetic, ONE_TERM, non-grouped, with EmptyDoc
+                Arguments.of(true, false, false, false, Set.of(syntheticKeyFor(1, 1001L, 1000L), syntheticKeyFor(2, 1002L, 999L))), // synthetic, ONE_TERM, non-grouped, no EmptyDoc
+                Arguments.of(true, true, true, true, Set.of(syntheticKeyFor(2, 1001L, 1000L), syntheticKeyFor(2, 1002L, 999L))), // synthetic, ALL_ENTRIES, grouped, with EmptyDoc
+                Arguments.of(true, true, true, false, Set.of(syntheticKeyFor(2, 1001L, 1000L))), // synthetic, ALL_ENTRIES, grouped, no EmptyDoc
+                Arguments.of(true, true, false, true, Set.of(syntheticKeyFor(1, 1001L, 1000L), syntheticKeyFor(2, 1002L, 999L), syntheticKeyFor(2, 1003L, 998L))), // synthetic, ALL_ENTRIES, non-grouped, with EmptyDoc
+                Arguments.of(true, true, false, false, Set.of(syntheticKeyFor(1, 1001L, 1000L), syntheticKeyFor(2, 1002L, 999L))) // synthetic, ALL_ENTRIES, non-grouped, no EmptyDoc
+        );
+    }
+
+    @ParameterizedTest(name = "indexScanTest({argumentsWithNames})")
+    @MethodSource("scanArguments")
+    public void indexScanTest(boolean isSynthetic, boolean matchAllDocs, boolean isGrouped, boolean includeEmptyDoc, Set<Tuple> expectedResult) throws Exception {
+        final long seed = 5363275763521L;
+        final LuceneIndexTestDataModel dataModel = new LuceneIndexTestDataModel.Builder(seed, this::getStoreBuilder, pathManager)
+                .setIsGrouped(isGrouped)
+                .setIsSynthetic(isSynthetic)
+                .setPrimaryKeySegmentIndexEnabled(true)
+                .setPartitionHighWatermark(10)
+                .build();
+
+        final long start = Instant.now().toEpochMilli();
+        // Populate data: 1 doc for groups 1 and 2 and one empty doc (if needed)
+        try (FDBRecordContext context = openContext()) {
+            dataModel.saveRecords(1, start, context, 1);
+            dataModel.saveRecords(1, start, context, 2);
```

Comment:
Using the same `start` doesn't seem right. Looking at the usages, I think `start` can just be pushed down into a field in `LuceneIndexTestModel`.

Reply:
Not sure what you have in mind. Is start meant to be unique (like `modSeq`)?